core/nandtool/nandtool.c

```diff
@@ -210,30 +242,38 @@ int main(int argc, char **argv)
 		nandtool_help(argv[0]);
 		return -EINVAL;
 	}
-	dev = argv[optind];
+
+	if ((dev = realpath(argv[optind], NULL)) == NULL) {
+		fprintf(stderr, "nandtool: failed to resolve path (%s): %s\n", argv[optind], strerror(errno));
+		return 1;
+	}
+
 
 	if ((err = lookup(dev, NULL, &nandtool_common.oid)) < 0) {
 		fprintf(stderr, "nandtool: failed to lookup device (%s), err: %d\n", dev, err);
-		return err;
+		free(dev);
+		return 1;
 	}
 
 	if ((err = open(dev, O_RDWR)) < 0) {
 		fprintf(stderr, "nandtool: failed to open device (%s), err: %d\n", dev, err);
-		return err;
+		free(dev);
+		return 1;
 	}
 	nandtool_common.fd = err;
 
 	if ((nandtool_common.info = flashmng_info(nandtool_common.oid)) == NULL) {
 		err = -EFAULT;
 		fprintf(stderr, "nandtool: failed to get device info (%s), err: %d\n", dev, err);
-		return err;
+		free(dev);
```

Comment:
Nitpick - missing `close(nandtool_common.fd);`

Reply:
not my code but added :)